src/Carnac.Logic/KeyProvider.cs

```diff
@@ -101,10 +106,19 @@ KeyPress ToCarnacKeyPress(InterceptKeyEventArgs interceptKeyEventArgs)
                            interceptKeyEventArgs.Key <= Keys.Z;
 
             var inputs = ToInputs(isLetter, winKeyPressed, interceptKeyEventArgs);
+            try
+            {
+                Icon icon = Icon.ExtractAssociatedIcon(process.MainModule.FileName);
```

Comment:
Maybe pull this and icon.ToImageSource into another static helper which keeps a dictionary of the images and only fetches if it doesn't exist?